faunadb-java-dsl/src/main/java/com/faunadb/client/types/FaunaConstructor.java

```diff
@@ -0,0 +1,43 @@
+package com.faunadb.client.types;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Instruct the decoder which constructor to use when decoding a object.
```

Comment:
Slightly better verbiage:

Specifies which constructor to use when decoding an object with a {@link Decoder}. You can also annotate a public static method, which will then be used instead of a constructor.